calico-vpp-agent/cni/network_vpp_routes.go

```diff
@@ -184,6 +198,143 @@ func (s *Server) CreatePodVRF(podSpec *storage.LocalPodSpec, stack *vpplink.Clea
+		// Always there (except multinet memif)
+		pathsToPod := []types.RoutePath{{
+			SwIfIndex: podSpec.TunTapSwIfIndex,
+			Gw:        containerIP.IP,
+		}}
+		// Add pbl memif case
+		if podSpec.MemifSwIfIndex != vpplink.INVALID_SW_IF_INDEX {
+			pathsToPod = append(pathsToPod, types.RoutePath{
+				SwIfIndex: podSpec.MemifSwIfIndex,
+				Gw:        containerIP.IP,
+			})
+			s.log.Infof("pod(add) add route to %+v in rpfvrf %+v via memif and tun", podSpec.GetContainerIps(), RPFvrfId)
+		} else {
+			s.log.Infof("pod(add) add route to %+v in rpfvrf %+v via tun", podSpec.GetContainerIps(), RPFvrfId)
+		}
+		route := &types.Route{
+			Dst:   containerIP,
+			Paths: pathsToPod,
+			Table: RPFvrfId,
+		}
+		err = s.vpp.RouteAdd(route)
+		if err != nil {
+			return errors.Wrapf(err, "error adding RPFVRF %d proper route", RPFvrfId)
+		} else {
+			stack.Push(s.vpp.RouteDel, route)
+		}
+
+		// Add addresses allowed to be spooofed
+		if podSpec.AllowedSpoofingPrefixes != "" {
+			// Parse Annotation data
+			allowedSources, err := s.ParseSpoofAddressAnnotation(podSpec.AllowedSpoofingPrefixes)
+			if err != nil {
+				return errors.Wrapf(err, "error parsing allowSpoofing addresses")
+			}
+			for _, allowedSource := range allowedSources {
+				s.log.Infof("pod(add) add route to %+v in rpfvrf %+v to allow spoofing", allowedSource.IPNet, RPFvrfId)
+				route := &types.Route{
+					Dst:   &allowedSource.IPNet,
+					Paths: pathsToPod,
+					Table: RPFvrfId,
+				}
+				err = s.vpp.RouteAdd(route)
+				if err != nil {
+					return errors.Wrapf(err, "error adding RPFVRF %d proper route", RPFvrfId)
+				} else {
+					stack.Push(s.vpp.RouteDel, route)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (s *Server) DeletePodRPFVRF(podSpec *storage.LocalPodSpec) error {
+	var err error
+	for _, containerIP := range podSpec.GetContainerIps() {
+		RPFvrfId := podSpec.GetRPFVrfId(vpplink.IpFamilyFromIPNet(containerIP))
+		// Always there (except multinet memif)
+		pathsToPod := []types.RoutePath{{
+			SwIfIndex: podSpec.TunTapSwIfIndex,
+			Gw:        containerIP.IP,
+		}}
+		// pbl memif case
+		if podSpec.MemifSwIfIndex != vpplink.INVALID_SW_IF_INDEX {
+			pathsToPod = append(pathsToPod, types.RoutePath{
+				SwIfIndex: podSpec.MemifSwIfIndex,
+				Gw:        containerIP.IP,
+			})
+			s.log.Infof("pod(del) del route to %+v in rpfvrf %+v via memif and tun", podSpec.GetContainerIps(), RPFvrfId)
+		} else {
+			s.log.Infof("pod(del) del route to %+v in rpfvrf %+v via tun", podSpec.GetContainerIps(), RPFvrfId)
+		}
+		err = s.vpp.RouteDel(&types.Route{
+			Dst:   containerIP,
+			Paths: pathsToPod,
+			Table: RPFvrfId,
+		})
+		if err != nil {
+			s.log.Errorf("error deleting RPFVRF %d route : %s", RPFvrfId, err)
+		}
+
+		// Delete addresses allowed to be spooofed
+		if podSpec.AllowedSpoofingPrefixes != "" {
+			// Parse Annotation data
+			allowedSources, err := s.ParseSpoofAddressAnnotation(podSpec.AllowedSpoofingPrefixes)
```

Comment:
We could maybe do this parsing in `ParsePodAnnotations())` and store only the parsed IPs in the `podSpec`. 
It might be a bit tricky as `LocalIPNet{}` types are not super clean. But I think failing early in case of a wrong annotation syntax would be nice.